src/core/server/workspaces/saved_objects/workspace_saved_objects_client_wrapper.ts

```diff
@@ -172,6 +172,9 @@ export class WorkspaceSavedObjectsClientWrapper {
     const findWithWorkspacePermissionControl = async <T = unknown>(
       options: SavedObjectsFindOptions
     ) => {
+      if (options.type === WORKSPACE_TYPE) {
+        return await wrapperOptions.client.find<T>(options);
+      }
       if (options.workspaces) {
         options.workspaces = options.workspaces.filter(
```

Comment:
hmm, I think `array.filter()` with `async/await` callback won't work as expected here, we need to use `for...of`

Reply:
Agree, this logic need to be changed when ACL class in ready.